src/pages/balances/BalancesPage.tsx

```diff
@@ -120,86 +118,196 @@ export function OtherBalanceChart() {
   )
 }
 
-export function BalanceText({ value, children }: PropsWithChildren<{ value: BalancesResponse["bitcoin"] }>) {
+interface BalanceDisplay {
+  amount: string
+  unit: string
+}
+
+export function BalanceText({ amount, unit, children }: PropsWithChildren<{ amount: string; unit: string }>) {
   return (
     <>
       <h3 className="scroll-m-20 text-2xl font-extrabold tracking-tight">
-        {value.value} {value.currency}
+        {amount} {unit}
       </h3>
       {children}
     </>
   )
 }
 
-function PageBody() {
-  const { data } = useSuspenseQuery({
-    queryKey: ["balances"],
-    queryFn: fetchBalances,
+function isErrorResponse<T>(res: { data?: T; error?: unknown }): res is { data: undefined; error: unknown } {
+  return res.error !== undefined
+}
```

Comment:
Is there a special reason to stop using tanstack query here? If it's suspense mode, I understand that it may be tricky to handle multiple queries at once with it, but you could still use the `useQueries` hook to call multiple functions at once, this way you wouldn't have to handle all the individual state updates and errors, and could still access the status imperatively with `isLoading`, `isError`, `isSuccess && data``, etc.

For reference:
https://tanstack.com/query/latest/docs/framework/react/reference/useQueries

Reference